docs/DeveloperGuide.md

```diff
@@ -5,9 +5,147 @@
+
+- Step 4: The calories record (`Entry`) stored in the `ArrayList<Entry> caloriesList` will be deleted by calling `calorieArrayList.remove((index-1));` and a successful deleting message will be shown in terminal by calling `CalorieListUi#successfulDeletedMessage(toDelete)`
+
+- Step 5: The latest calories list will be updated to saving file by calling `CalorieList#updateFile()`.
+
+The Class diagram for Calories delete feature is shown below:
+
+![CaloriesDeleteClassDiagram](https://github.com/a-wild-chocolate/tp/blob/master/docs/caloriesDeleteUML.jpg)
+
+### Parsing user input for hydration entries
+
+This functionality is facilitated by `ParserHydration`. It implements one operation, namely:
+- `ParserHydration#parseHydrationInput(String input)`
+
+This operation is exposed in the `HydrationList` class as `HydrationList#addEntry(String)`.
+
+Given below is an example usage scenario and how this mechanism behaves at every step:
+- Step 1: When the user inputs the command `hydration in Milo v/100 d/2022-01-02` in the terminal,
+  the string is sent to `HydrationList#addEntry(String)`, which calls `ParserHydration#parseHydrationInput(String)`.
+
+- Step 2: Using `String.split()`, the method extracts information such as the description, volume of beverage, and date of entry. The obtained information is sent to the private method `ParserHydration#makeNewInputEntry(String, int, String)` to create a new entry of class `HydrationEntry` that extends `Entry`.
+
+- Step 3: The created `HydrationEntry` instance is added into the `ArrayList<Entry>` attribute of the `HydrationList`.
+
+### Calculating hydration requirements for each user
+
+#### Design Considerations
+
+**General Health Guidelines:** The recommended daily intake of water for an average adult is around 8 glasses or approximately 2000 milliliters. This guideline is commonly recommended by health authorities and organizations such as HealthHub.
+
+**Ease of Implementation:** Setting a standard hydration requirement simplifies the tracking process for users. It provides a clear goal to strive for, making it easier for individuals to monitor and maintain their hydration levels consistently.
+### Hydration list feature
+
+The `hydration list` feature lists out the record of all the Hydration data that the user has keyed in. The Hydration data are all stored into a `ArrayList<Entry> hydrationArrayList` attribute of the `HydrationList` Class. Hydration data are printed when the `printHydrationList()` function is called.
+
+The `printHydrationList()` function iterates through the `hydrationArrayList` and prints out the Entries according to its order in the Array List.
+
+The Class diagram for Hydration list feature is shown below. Unrelated attributes and Classes were excluded.
+
+![HydrationListClassDiagram.png](HydrationListClassDiagram.png)
```

Comment:
The association between hydration and hydrationlist does not seem to be a notation used in cs2113. Would it be better to update the diagram to use association lines instead?

Reply:
Yeah, this makes sense here.